src/movies.js

```diff
@@ -1,16 +1,79 @@
-// Iteration 1: All directors? - Get the array of all directors.
-// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
+//Iteration 1
+function getAllDirectors(movies) {
+  const directorsArray = movies.map(function (movies) {
+    return movies.director;
+  });
+  return directorsArray;
+}
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+//BONUS 1.1
 
-// Iteration 3: All rates average - Get the average of all rates with 2 decimals
+//Iteration 2
+function howManyMovies(movies) {
+  const spielbergsDramas = movies.filter(function (movies) {
+    return (
+      movies.director === "Steven Spielberg" && movies.genre.includes("Drama")
+    );
+  });
+  return spielbergsDramas.length;
+}
 
-// Iteration 4: Drama movies - Get the average of Drama Movies
+//Iteration 3
+function ratesAverage(movies) {
+  if (movies.length === 0) {
+    return 0;
+  }
+  let totalRate = movies.reduce(function (acc, val) {
+    if (!val.rate) {
+      return acc;
+    }
+    return acc + val.rate;
+  }, 0);
+  return parseFloat((totalRate / movies.length).toFixed(2));
+}
 
-// Iteration 5: Ordering by year - Order by year, ascending (in growing order)
+//Iteration 4
+function dramaMoviesRate(movies) {
+  const onlyDramas = movies.filter(function (movies) {
+    return movies.genre.includes("Drama");
+  });
+  if (onlyDramas.length === 0) {
+    return 0;
+  }
+  const totalDramasRate = onlyDramas.reduce(function (acc, val) {
+    return acc + val.rate;
+  }, 0);
+  return parseFloat((totalDramasRate / onlyDramas.length).toFixed(2));
```

Comment:
You could also have used the previous function "ratesAverage" to calculate the drama movies average here! 💡